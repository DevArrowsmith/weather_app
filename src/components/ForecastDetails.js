import React from "react";
import PropTypes from "prop-types";
import WeatherIcon from "react-icons-weather";
import moment from "moment";
import "../styles/ForecastDetails.css";

const ForecastDetails = ({ forecast }) => {
    const { date, description, humidity, icon, temperature, wind } = forecast;
    return (
        <div className="forecast-details" data-testid="forecast-details">
            <div className="forecast-details__date">
                {moment(date).format("ddd Do MMM")}
            </div>
            <div className="forecast-details__description">{description}</div>
            <div className="forecast-details__icon" data-testid="forecast-icon">
                <WeatherIcon name="owm" iconId={icon.toString()} />
            </div>
            <div className="forecast-details__temperature-max">
                Maximum temperature: {temperature.max}°C
            </div>
            <div className="forecast-details__temperature-min">
                Minimum temperature: {temperature.min}°C
            </div>
            <div className="forecast-details__humidity">
                Humidity: {humidity}g/m3
            </div>
            <div className="forecast-details__wind-speed">
                Wind speed: {wind.speed}mph
            </div>
            <div className="forecast-details__wind-direction">
                Wind direction: {wind.direction}
            </div>
        </div>
    );
};

ForecastDetails.propTypes = {
    forecast: PropTypes.shape({
        date: PropTypes.number,
        description: PropTypes.string,
        humidity: PropTypes.number,
        icon: PropTypes.number,
        temperature: PropTypes.shape({
            max: PropTypes.number,
            min: PropTypes.number,
        }),
        wind: PropTypes.shape({
            direction: PropTypes.string,
            speed: PropTypes.number,
        }),
    }).isRequired,
};

export default ForecastDetails;
