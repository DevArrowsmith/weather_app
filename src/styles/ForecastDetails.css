.forecast-details {
    margin-bottom: 20px;
}