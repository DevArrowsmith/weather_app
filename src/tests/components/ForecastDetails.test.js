import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../../components/ForecastDetails";

describe("ForecastDetails", () => {
    const validProps = {
        forecast: {
            date: 1525046400000,
            description: "Stub description",
            humidity: 30,
            icon: "800",
            temperature: {
                min: 12,
                max: 34,
            },
            wind: {
                speed: 20,
                direction: "s",
            },
        },
    };

    it("renders correctly", () => {
        const { asFragment } = render(
            <ForecastDetails forecast={validProps.forecast} />
        );
        expect(asFragment()).toMatchSnapshot();
    });

    it("renders using the correct props", () => {
        const { getByText, getByTestId } = render(
            <ForecastDetails forecast={validProps.forecast} />
        );

        expect(getByText("Mon 30th Apr")).toHaveClass("forecast-details__date");
        expect(getByText("Stub description")).toHaveClass(
            "forecast-details__description"
        );
        expect(getByText("Humidity: 30g/m3")).toHaveClass(
            "forecast-details__humidity"
        );
        expect(getByTestId("forecast-icon")).toHaveClass(
            "forecast-details__icon"
        );
        expect(getByText("Maximum temperature: 34°C")).toHaveClass(
            "forecast-details__temperature-max"
        );
        expect(getByText("Minimum temperature: 12°C")).toHaveClass(
            "forecast-details__temperature-min"
        );
        expect(getByText("Wind speed: 20mph")).toHaveClass(
            "forecast-details__wind-speed"
        );
        expect(getByText("Wind direction: s")).toHaveClass(
            "forecast-details__wind-direction"
        );
    });
});
