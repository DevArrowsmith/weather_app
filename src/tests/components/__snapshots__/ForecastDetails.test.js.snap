// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
    data-testid="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      Mon 30th Apr
    </div>
    <div
      class="forecast-details__description"
    >
      Stub description
    </div>
    <div
      class="forecast-details__icon"
      data-testid="forecast-icon"
    >
      <i
        class="wi wi-day-sunny"
        name="owm"
      />
    </div>
    <div
      class="forecast-details__temperature-max"
    >
      Maximum temperature: 34°C
    </div>
    <div
      class="forecast-details__temperature-min"
    >
      Minimum temperature: 12°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      Humidity: 30g/m3
    </div>
    <div
      class="forecast-details__wind-speed"
    >
      Wind speed: 20mph
    </div>
    <div
      class="forecast-details__wind-direction"
    >
      Wind direction: s
    </div>
  </div>
</DocumentFragment>
`;
